File: libs/event-bus/src/topics.rs
/// Well-known NATS subject prefixes for each service domain.
pub mod subjects {
    pub const AUTH: &str = "of.auth";
    pub const DATASETS: &str = "of.datasets";
    pub const DATASET_QUALITY: &str = "of.datasets.quality";
    pub const PIPELINES: &str = "of.pipelines";
    pub const WORKFLOWS: &str = "of.workflows";
    pub const ONTOLOGY: &str = "of.ontology";
    pub const QUERIES: &str = "of.queries";
    pub const AUDIT: &str = "of.audit";
    pub const NOTIFICATIONS: &str = "of.notifications";
}

/// JetStream stream names.
pub mod streams {
    pub const EVENTS: &str = "OF_EVENTS";
    pub const AUDIT: &str = "OF_AUDIT";
    pub const NOTIFICATIONS: &str = "OF_NOTIFICATIONS";
}

File: libs/event-bus-control/topics.go
package controlbus

// Subject prefixes — kept verbatim with the Rust workspace so cross-
// language services route to the same hierarchy.
const (
	SubjectAuth           = "of.auth"
	SubjectDatasets       = "of.datasets"
	SubjectDatasetQuality = "of.datasets.quality"
	SubjectPipelines      = "of.pipelines"
	SubjectWorkflows      = "of.workflows"
	SubjectOntology       = "of.ontology"
	SubjectQueries        = "of.queries"
	SubjectAudit          = "of.audit"
	SubjectNotifications  = "of.notifications"
)

// JetStream stream names.
const (
	StreamEvents        = "OF_EVENTS"
	StreamAudit         = "OF_AUDIT"
	StreamNotifications = "OF_NOTIFICATIONS"
)

// Specific event-type / subject constants — match Rust contracts.rs.
const (
	DatasetQualityRefreshRequestedEventType = "dataset.quality.refresh.requested"
	DatasetQualityRefreshRequestedSubject   = "of.datasets.quality.refresh.requested"

	WorkflowTriggerRequestedEventType = "workflow.trigger.requested"
	WorkflowTriggerRequestedSubject   = "of.workflows.trigger.requested"

	NotificationEventType = "notification.updated"
	NotificationSubject   = "of.notifications.updated"
)
